core/src/main/java/org/apache/carbondata/core/metadata/schema/table/TableSchema.java

```diff
@@ -77,6 +78,7 @@
 
   public TableSchema() {
     this.listOfColumns = new ArrayList<ColumnSchema>(CarbonCommonConstants.DEFAULT_COLLECTION_SIZE);
+    this.tableProperties = new HashMap<String, String>(5);
```

Comment:
I think it is ok to use default constructor of HashMap, 5 is not needed

Reply:
yeah, I just follow the previous line to set a default value and think 16 is too big. Sure to fix it.